### PR TITLE
treewide: modify Kconfig symbol prefixes

### PR DESCRIPTION
**Pull request by leandrolanzieri**
### Contribution description
It was proposed by @aabadie to modify the current prefix conventions for Kconfig symbols while we are in migration, to reduce possible ambiguities. After some [discussion](https://github.com/RIOT-OS/RIOT/pull/14626#discussion_r475079120), the proposal is the following:

| Current prefix | New prefix | Symbol description |
| ------------------ | -------------- | -------------------------- |
| `MODULE_` | `USEMODULE_` | Symbols generated from `USEMODULE` variable. Needed only during migration. |
| `KCONFIG_MODULE_` | `KCONFIG_USEMODULE_` | Symbols used in `menuconfig`s to allow optional configuration of used modules. Depend on `USEMODULE_` symbols. |
| `PKG_` | `USEPKG_` | Symbols generated from `USEPKG` variable. Needed only during migration. |
| `KCONFIG_PKG_` | `KCONFIG_USEPKG_` | Symbols used in `menuconfig`s to allow optional configuration of used packages. Depend on`USEPKG_` symbols. |
| `MOD_` | `MODULE_` | Symbols that model modules in Kconfig. |
| -- (no packages modelled yet) | `PACKAGE_` | Symbols that model packages in Kconfig. |

### Testing procedure
- Check that the naming convention makes sense and is understandable.
- Configurations should be shown and applied as usual (the `USEMODULE` and `USEPKG` variables should generate the same symbols which activate the visibility of configurations).
- Check that all symbols have been renamed and documentation has been updated, according the table above.
- Green `CI`.

### Issues/PRs references
Suggested in #14626

### CODE REVIEW

File: examples/gnrc_border_router/Kconfig [hunk 1]
```diff
@@ -1,10 +1,10 @@
-if MODULE_ETHOS
+if USEMODULE_ETHOS
 config GNRC_DHCPV6_CLIENT_6LBR_STATIC_ROUTE
     default y
-    depends on MODULE_GNRC_DHCPV6_CLIENT_6LBR && KCONFIG_MODULE_GNRC_DHCPV6
+    depends on USEMODULE_GNRC_DHCPV6_CLIENT_6LBR && KCONFIG_USEMODULE_GNRC_DHCPV6
```
Comment (aabadie) [inline examples/gnrc_border_router/Kconfig:4]: Should it rather be:
```suggestion
    depends on USEMODULE_GNRC_DHCPV6_CLIENT_6LBR && USEMODULE_GNRC_DHCPV6
```
(which means there's an issue in master).

File: examples/gnrc_border_router/Kconfig [hunk 2]
```diff
@@ -1,10 +1,10 @@
...
 config GNRC_NETIF_IPV6_ADDRS_NUMOF
     # CONFIG_GNRC_DHCPV6_CLIENT_6LBR_STATIC_ROUTE=1 requires one more address
     # for `fe80::2`.
     default 3
-    depends on KCONFIG_MODULE_GNRC_NETIF
-endif # MODULE_ETHOS
+    depends on KCONFIG_USEMODULE_GNRC_NETIF
```
Comment (aabadie) [inline examples/gnrc_border_router/Kconfig:9]: Same here:
```suggestion
    depends on USEMODULE_GNRC_NETIF
```
?

File: sys/pm_layered/Kconfig [hunk 3]
```diff
@@ -5,7 +5,7 @@
 # directory for more details.
 #
 
-config MOD_PM_LAYERED
+config MODULE_PM_LAYERED
     bool "Platform-independent Power Management"
-    depends on MOD_PERIPH_PM
+    depends on MODULE_PERIPH_PM
```
Comment (aabadie) [inline sys/pm_layered/Kconfig:10]: Shouldn't this rather be:
```suggestion
    depends on USEMODULE_PERIPH_PM
```
?

File: tests/gnrc_dhcpv6_client_6lbr/Kconfig [hunk 4]
```diff
@@ -1,10 +1,10 @@
-if MODULE_ETHOS
+if USEMODULE_ETHOS
 config GNRC_DHCPV6_CLIENT_6LBR_STATIC_ROUTE
     default y
-    depends on MODULE_GNRC_DHCPV6_CLIENT_6LBR && KCONFIG_MODULE_GNRC_DHCPV6
+    depends on USEMODULE_GNRC_DHCPV6_CLIENT_6LBR && KCONFIG_USEMODULE_GNRC_DHCPV6
```
Comment (aabadie) [inline tests/gnrc_dhcpv6_client_6lbr/Kconfig:4]:
```suggestion
    depends on USEMODULE_GNRC_DHCPV6_CLIENT_6LBR && USEMODULE_GNRC_DHCPV6
```
?

File: tests/gnrc_dhcpv6_client_6lbr/Kconfig [hunk 5]
```diff
@@ -1,10 +1,10 @@
...
 config GNRC_NETIF_IPV6_ADDRS_NUMOF
     # CONFIG_GNRC_DHCPV6_CLIENT_6LBR_STATIC_ROUTE=1 requires one more address
     # for `fe80::2`.
     default 3
-    depends on KCONFIG_MODULE_GNRC_NETIF
-endif # MODULE_ETHOS
+    depends on KCONFIG_USEMODULE_GNRC_NETIF
```
Comment (aabadie) [inline tests/gnrc_dhcpv6_client_6lbr/Kconfig:9]:
```suggestion
    depends on USEMODULE_GNRC_NETIF
```
?

File: tests/gnrc_ipv6_ext_frag/app.config [hunk 6]
```diff
@@ -1,3 +1,3 @@
 # This test fails if the pool size is less than 3
-CONFIG_KCONFIG_MODULE_GNRC_IPV6_EXT_FRAG=y
+CONFIG_KCONFIG_USEMODULE_GNRC_IPV6_EXT_FRAG=y
```
Comment (aabadie) [inline tests/gnrc_ipv6_ext_frag/app.config:2]:
```suggestion
CONFIG_MODULE_GNRC_IPV6_EXT_FRAG=y
```
?

**Comment by aabadie** [comment]
I have some comments, not strictly related to this PR but I'm wondering if there aren't some issues in master.

File: examples/gnrc_border_router/Kconfig (hunk 1 above)
Comment (leandrolanzieri) [reply examples/gnrc_border_router/Kconfig:4]: No, `KCONFIG_USEMODULE_GNRC_DHCPV6` is a dependency so the value is only applied when the configuration via Kconfig for that module is enabled.

File: examples/gnrc_border_router/Kconfig (hunk 2 above)
Comment (leandrolanzieri) [reply examples/gnrc_border_router/Kconfig:9]: See above

File: sys/pm_layered/Kconfig (hunk 3 above)
Comment (leandrolanzieri) [reply sys/pm_layered/Kconfig:10]: No, both symbols (`MODULE_PM_LAYERED` and `MODULE_PERIPH_PM`) are part of the Kconfig module migration (that's why they depend on `TEST_KCONFIG` as well).

File: tests/gnrc_dhcpv6_client_6lbr/Kconfig (hunk 4 above)
Comment (leandrolanzieri) [reply tests/gnrc_dhcpv6_client_6lbr/Kconfig:4]: Same as the other

File: tests/gnrc_dhcpv6_client_6lbr/Kconfig (hunk 5 above)
Comment (leandrolanzieri) [reply tests/gnrc_dhcpv6_client_6lbr/Kconfig:9]: Same as the other

File: tests/gnrc_ipv6_ext_frag/app.config (hunk 6 above)
Comment (leandrolanzieri) [reply tests/gnrc_ipv6_ext_frag/app.config:2]: Here we are activating the 'configurability' of the module via Kconfig.

**Review by aabadie — APPROVED** [review]
Thanks for the clarifications. I couldn't find any problem so far. The new names seems more consistent to me.

ACK

**Comment by aabadie** [comment]
@cgundogan, what do you think of the new naming scheme ?

**Review by cgundogan — APPROVED** [review]
> @cgundogan, what do you think of the new naming scheme ?>@cgundogan, what do you think of the new naming scheme ?

my honest opinion? (: ..

Nevertheless, the changes look good. There are a lot of changes here .. but since most of them come from a systematic search & replace, the margin for careless slips should be rather small.

**Pull request merged**